scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/InMemoryConnectionStore.java

```diff
@@ -143,6 +143,10 @@ public InMemoryConnectionStore(final int capacity, final long threshold, final S
 
 				@Override
 				public void onEviction(Connection staleConnection) {
+					Handshaker handshaker = staleConnection.getOngoingHandshake();
+					if (handshaker != null) {
+						handshaker.handshakeFailed(new Exception("Connection evicted from store"));
+					}
```

Comment:
Should this change be integrated in 2.0.x ? (I mean even if we don't integrate this PR?)

Reply:
I want to execute this in a serialzed executor, not to break this at the very end.
And I would prefer to call this as timeout much earliere, as I already mentioned in #343.

Reply:
FMPOV, it would wire up the serial execution, if we pass it into the connection store.
With the move of the flight and the introduction of the timeout, handshakes should not be ongoing, when connections get evicted.